includes/wc-core-functions.php

```diff
@@ -150,6 +150,86 @@ function wc_update_order( $args ) {
 	return wc_create_order( $args );
 }
 
+/**
+ * Given a path, this will convert any of the subpaths into their corresponding tokens.
+ *
+ * @since 4.1.0
+ * @param string $path The absolute path to tokenize.
+ * @param array  $path_tokens An array keyed with the token, containing paths that should be replaced.
+ * @return string The tokenized path.
+ */
+function wc_tokenize_path( $path, $path_tokens ) {
+	// Order most to least specific so that the token can encompass as much of the path as possible.
+	uasort(
+		$path_tokens,
+		function ( $a, $b ) {
+			$a = strlen( $a );
+			$b = strlen( $b );
+
+			if ( $a > $b ) {
+				return -1;
+			}
+
+			if ( $b > $a ) {
+				return 1;
+			}
+
+			return 0;
+		}
+	);
+
+	foreach ( $path_tokens as $token => $token_path ) {
+		if ( 0 !== strpos( $path, $token_path ) ) {
+			continue;
+		}
+
+		$path = str_replace( $token_path, '{{' . $token . '}}', $path );
+	}
+
+	return $path;
+}
+
+/**
+ * Given a tokenized path, this will expand the tokens to their full path.
+ *
+ * @since 4.1.0
+ * @param string $path The absolute path to expand.
+ * @param array  $path_tokens An array keyed with the token, containing paths that should be expanded.
+ * @return string The absolute path.
+ */
+function wc_untokenize_path( $path, $path_tokens ) {
+	foreach ( $path_tokens as $token => $token_path ) {
+		$path = str_replace( '{{' . $token . '}}', $token_path, $path );
+	}
+
+	return $path;
+}
+
+/**
+ * Fetches an array containing all of the configurable path constants to be used in tokenization.
+ *
+ * @return array The key is the define and the path is the constant.
+ */
+function wc_get_path_define_tokens() {
+	$defines = array(
+		'ABSPATH',
+		'WC_ABSPATH',
+		'WP_CONTENT_DIR',
+		'WP_PLUGIN_DIR',
+		'PLUGINDIR',
+		'WP_THEME_DIR',
+	);
+
+	$path_tokens = array();
+	foreach ( $defines as $define ) {
+		if ( defined( $define ) ) {
+			$path_tokens[ $define ] = constant( $define );
+		}
+	}
+
+	return apply_filters( 'wc_get_path_define_tokens', $path_tokens );
```

Comment:
I went with the `wc_` prefix because it was consistent with all of the other template return overrides, which simply share the name of the function. Do you think that consistency might be more important than convention here?

Reply:
There's no consistent using `wc_` for filter names, in fact any filter or hook using `wc_` is incorrect and should get deprecated or removed in next versions.
But for some reasons we keep accepting 3rd party Pull Request with incorrect prefixes.

Reply:
We have it also documented on: https://github.com/woocommerce/woocommerce/wiki/Naming-conventions

Reply:
Sorry, I was referring to `wc_get_template` and `wc_get_template_part`, but I do prefer the more correct prefix :)